File: BAEKJOON/Javascript/25000/25300/25304.js
/*
Title : 영수증
Link : https://www.acmicpc.net/problem/25304
*/

const fs = require("fs");
const filepath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt";
const input = fs.readFileSync(filepath).toString().trim().split("\n");

const X = Number(input[0]);
const N = Number(input[1]);
let totalPrice = 0;
for (let i = 0; i < N; i++) {
  const [price, count] = input[i + 2].split(" ").map(Number);
  totalPrice += price * count;
}
console.log(totalPrice === X ? "Yes" : "No");

File: BAEKJOON/Python/25000/25300/25304.py
"""
Title : 영수증
Link : https://www.acmicpc.net/problem/25304
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    X = int(input())
    total_price = 0
    for _ in range(int(input())):
        price, count = map(int, input().split())
        total_price += price * count
    print("Yes" if total_price == X else "No")
